### ISSUE TITLE
incorrect timestamps on files checked out by git

### ISSUE THREAD

**Issue by adriancole** [travis-ci/travis-ci#7254]
After investigation, it seems like files are getting the wrong timestamps

Ex I added this into the .travis.yml
```
install:
  - date
  - git --version
  - git log -n 1 ./zipkin/src/test/java/zipkin/internal/DependencyLinkerTest.java
```

then the output for a file was this..

```
$ date
Thu Feb  2 04:13:46 UTC 2017
install.2
0.03s$ git --version
git version 2.11.0
install.3
0.02s$ git log -n 1 ./zipkin/src/test/java/zipkin/internal/DependencyLinkerTest.java
commit ffac4ff4eac53b82dea2f66179e3be920bfa6276
Author: Adrian Cole <adriancole@users.noreply.github.com>
Date:   Tue Jan 3 09:28:48 2017 +0800
```

you can see that it says the file was modified on Jan 3 this year..

however, it wasn't.. it was modified last year.

```bash
$ git log -n 1 ./zipkin/src/test/java/zipkin/internal/DependencyLinkerTest.java
commit 21fc32b965e70f0985a164f34ed89ae55e115a11
Author: Adrian Cole <adriancole@users.noreply.github.com>
Date:   Thu Oct 27 09:38:34 2016 +0800
```

This is causing havoc on our license check plugin https://github.com/openzipkin/zipkin/issues/1512

**Comment by BanzaiMan** [comment]
Where do you see the "Oct 27" output?

**Comment by adriancole** [comment]
I saw it in the travis log (after I added commands into my .travis file so
that they could show up)

https://travis-ci.org/openzipkin/zipkin/builds/197529456

**Comment by BanzaiMan** [comment]
This is fascinating. openzipkin/zipkin@ffac4ff4eac53b82dea2f66179e3be920bfa6276 does not touch this file at all, and at that commit, the file in question should have the October 27 time stamp: https://github.com/openzipkin/zipkin/blob/ffac4ff/zipkin/src/test/java/zipkin/internal/DependencyLinkerTest.java

**Comment by adriancole** [comment]
yeah I've not seen anything like that before either.

**Comment by BanzaiMan** [comment]
Maybe you could introduce an artificial change (say, a blank line, a spurious comment, an extra white space) to the file and see if the time stamp is updated.

**Comment by sjoerdtalsma** [comment]
@adriancole do you know whether this issue got resolved at all?

I'm seeing the exact same thing on my repo, travis telling me that files were last-changed in a commit that only updated a pom.xml dependency.

e.g. https://travis-ci.org/talsma-ict/reflection/builds/369983327

**Comment by sjoerdtalsma** [comment]
> no it didn't resolve

Thanks for the update!
:disappointed: I just updated the copyright year in all my java files, unfortunate hassle

**Comment by pavolloffay** [comment]
I had the same issue. In my case git on travis was showing a wrong history which is also the case in @adriancole first comment. It also shows (grafted) in newer git versions. If you replicate how travis clones the repo you should get the same result locally.

I have solved it by:

```yml
git:
  depth: false
```

**Comment by pavolloffay** [comment]
indeed, I am just cross-linking related issues https://github.com/travis-ci/travis-ci/issues/7254#issuecomment-387689631 https://github.com/openzipkin/zipkin/issues/1512
